__tests__/app.test.js

```diff
@@ -162,7 +162,83 @@ describe('GET /api/articles/:article_id/comments', () => {
         })
     })
   })
-  
+
+
+describe("POST: /api/articles/2/comments", () => {
+    test("status 201: should post a comment for an article and respond with the posted comment", () => {
+        const data = {
+            body: "This is a great article",
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.comment).toMatchObject({
+                article_id:expect.any(Number),
+                author:expect.any(String),
+                body:expect.any(String),
+                comment_id:expect.any(Number),
+                created_at:expect.any(String),
+                votes:expect.any(Number)
+            })
+        })
+    })
+    test("status 400: no username inputted results in error message", () => {
+        const data = {
+            body: "This is a great article"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Request is missing key information")
+        })
+    })
+    test("status 400: incorrect data type used for username results in error message", () => {
+        const data = {
+            username: 123,
+            body: "Great article"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("One or more inputs incorrectly formatted in request")
```

Comment:
A side note!

If you are fed up of making new messages it is perfectly fine to simply respond with "bad request" for all these bad request errors. We don't have to be _that_ helpful or specific as there will be/is documentation to show how to use the api. Sending "bad request" is putting the ball in the court of the client to use the api properly and means as a dev we don't have to cover each error with unique messages! (Also it is possible to give information about our database if we get too specific which we don't want to do)

Reply:
That's good to know!